### ISSUE TITLE
eth_newFilter returns malformed ids with trailing zeroes

### ISSUE THREAD

**Issue by Steffen93** [trufflesuite/ganache-core#58]
## Expected Behavior
When a new filter is created using eth_newFilter, the ID returned should be formed correctly according to [this description on how to encode quantities](https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding), i.e., no trailing zeroes like "0x**0**3"

## Current Behavior
The ID returned after filter creation contains leading zeroes like "0x03" instead of "0x3", also only accepting the same string when the filter changes are queried. So querying filter changes for "0x3" will result in the error message "FilterSubprovider - no filter with that id: 0x3". So, if the ID is stored as an integer and (de-)serialized before calls are made, the previous error occurs.

## Context
I stumbled upon the problem using the [rust-web3 library](https://github.com/tomusdrw/rust-web3) and also opened an [issue](https://github.com/tomusdrw/rust-web3/issues/87) there which resulted in a temporary workaround. Nevertheless I think the root cause is here and it should be fixed.

## My Environment

* Version used: Ganache v1.0.2 / ganache-cli@6.0.3
* Operating System and version: Ubuntu 16.04.3 LTS

**Comment by jasonadkison** [comment]
I'm experiencing the same issue using Ganache CLI v6.1.0-beta.0 (ganache-core: 2.1.0-beta.0)

**Comment by bh2smith** [comment]
Has anything been done regarding this issue?!

I am also getting back result  from eth_getLogs with 

```
"result":[{"logIndex":"0x00","transactionIndex":"0x00"...
```

which appears to be coming from 

ganache-core/lib/utils/to.js at line 59, where

``` js
if(val.length % 2 != 0) {
    val = "0" + val;
}
```

According to the ethereum wiki for [JSON-RPC](https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding) for allowed vs not allowed encoded QUANTITIES (including **logIndex** and **transactionindex**), the result dict should really be more like

```
"result":[{"logIndex":"0x0","transactionIndex":"0x0"...
```

Notice that the problem with this response lies within [go-ethereum/common/hexutil/json.go](https://github.com/ethereum/go-ethereum/blob/master/common/hexutil/json.go) at line 306

``` go
if len(input) > 1 && input[0] == '0' {
    return nil, ErrLeadingZero
}
```

Which throws a leading zero error for such response values

**Comment by eshaben** [comment]
@bh2smith, I'm sorry that I didn't respond with my findings before this issue was closed! It looks like this was fixed by changing the way we handled the hex encoding of the `logIndex` and `transactionIndex` values. The changes were made in this commit: https://github.com/trufflesuite/ganache-core/commit/d187a66eabc1d85c692c4f9fe26f69dacac48ab1. I validated the changes through the tests created in this commit: https://github.com/trufflesuite/ganache-core/commit/45301c2679c942ddbfa932e6e31c224b18e609a5.
